### PR TITLE
[GraphQL] Use correct resource configuration for filter args of nested collection

### PR DESCRIPTION
**Pull request by alanpoulain**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #2969
| License       | MIT
| Doc PR        |

Backport of https://github.com/api-platform/core/pull/2970

### CODE REVIEW

**Comment by teohhanhui** [comment]
Should it be backported? It's a bug, but users may have come to rely on the buggy behaviour, no?

**Comment by alanpoulain** [comment]
I don't think so. The schema was generated with the wrong filters and you couldn't use them.

File: features/graphql/filters.feature [hunk 1]
```diff
@@ -107,11 +107,33 @@ Feature: Collections filtering
       }
     }
     """
-    And the JSON node "data.dummies.edges[0].node.relatedDummies.edges" should have 0 elements
-    And the JSON node "data.dummies.edges[1].node.relatedDummies.edges" should have 0 elements
+    Then the JSON node "data.dummies.edges[0].node.relatedDummies.edges" should have 0 element
```
Comment (teohhanhui) [inline features/graphql/filters.feature]: Actually, zero is always followed by the plural form in English.

Comment (alanpoulain) [reply features/graphql/filters.feature]: Ho yes you're right. But it's in master now, there will be some conflicts. Do you want to keep them anyway?

Comment (teohhanhui) [reply features/graphql/filters.feature]: We can resolve the conflicts when merging back to `master`.

**Pull request merged**